service/history/workflow/mutable_state_impl.go

```diff
@@ -649,6 +646,32 @@ func (ms *MutableStateImpl) TaskQueue() *taskqueuepb.TaskQueue {
 	}
 }
 
+func (ms *MutableStateImpl) SetStickyTaskQueue(name string, scheduleToStartTimeout *time.Duration) {
+	ms.executionInfo.StickyTaskQueue = name
+	ms.executionInfo.StickyScheduleToStartTimeout = scheduleToStartTimeout
+}
+
+func (ms *MutableStateImpl) ClearStickyTaskQueue() {
+	ms.executionInfo.StickyTaskQueue = ""
+	ms.executionInfo.StickyScheduleToStartTimeout = nil
+}
+
+// TaskQueueScheduleToStartTimeout returns TaskQueue struct and corresponding StartToClose timeout.
+// Task queue kind (sticky or normal) and timeout are set based on comparison of normal task queue name
+// in mutable state and provided name.
+func (ms *MutableStateImpl) TaskQueueScheduleToStartTimeout(name string) (*taskqueuepb.TaskQueue, *time.Duration) {
```

Comment:
nit: I would return by value here rather than pointers but I understand that's not the common pattern.

Reply:
That's because of proto. Both return values goes to proto and there will be `&` operator anyway. Doing it right away simplifies caller code.